src/Payments/GPay.res

```diff
@@ -33,6 +33,14 @@ let make = (
 
   let areOneClickWalletsRendered = Recoil.useSetRecoilState(RecoilAtoms.areOneClickWalletsRendered)
 
+  let isGuestCustomer = React.useMemo1(() => {
+    switch options.customerPaymentMethods {
+    | LoadedSavedCards(_, false)
+    | NoResult(false) => false
+    | _ => true
+    }
+  }, [options.customerPaymentMethods])
+
```

Comment:
useMemo is required ?

Reply:
For GuestCustomer it's done in this way. I'll refactor and store the customer/payment_method response in recoil for further use. I haven't done in this PR because it will increase unnecessary files.


That's why duplicate functions I have written. I'll refactor this guest customer logic once this PR gets closed

Reply:
Updated.